### ISSUE TITLE
Make diagram accessible for screen readers (blind and visually impared)

### ISSUE THREAD

**Issue by hamidrezaabroshan** [bpmn-io/bpmn-js#1548]
Hi,
I filed <a href="https://github.com/camunda-community-hub/zeebe-simple-monitor/issues/334#issuecomment-986790764"> this bug </a> and they said that the bug relates to bpmn.io.

As you probably can guess, the canvas html element is not accessible  by default for screen readers because they can only read textual information.
As this component may used several time in zeebe simple monitor app, fixing this component can fix several issues.
To make canvas accessible, you can google making canvas accessible for screen readers and for instance, you can look at <a href="https://stevefaulkner.github.io/Articles/Notes%20on%20accessibility%20of%20text%20replacement%20using%20HTML5%20canvas.html"> this article </a>
Also, you can test accessibility of components by using free and open source <a href="https://www.nvaccess.org/download/"> nvda screen reader </a> on windows.
You should be able to navigate in your page by arrow keys and tab or shift+tab.
and be able to findout this diagram info.

Thanks.

---

A good chunk of additional context on A11y compliant diagram navigation is provided [here](https://forum.bpmn.io/t/how-to-make-modeler-keyboard-accessible/5996/5?u=nikku).

---

Depends on https://github.com/bpmn-io/diagram-js/issues/567.

**Comment by nikku** [comment]
@hamidrezaabroshan Which exact information would you like to have available for each flow node?

I believe you filed a very similar issue with https://github.com/camunda/camunda-modeler/issues/2588, right?

What shall be different in this particular case?

**Comment by hamidrezaabroshan** [comment]
@nikku,
I think that this issue is for different app.
this issue is for <a href="https://github.com/camunda-community-hub/zeebe-simple-monitor"> zeebe simple monitor</a> and that issue is fore <a href="https://github.com/camunda/camunda-modeler">camunda-modeler </a>

Also, I think solving this issue is different and simpler.
If you think that these are same, please comment down below or if you think that this issue does not relate to this repo please comment <a href="https://github.com/camunda-community-hub/zeebe-simple-monitor/issues/334#issuecomment-986790764"> here </a>
Thanks.

**Comment by hamidrezaabroshan** [comment]
In simple monitore, /views/instances/{key}  page shows a diagram that indicates which flow is executed and which has errors.
I just want name of flows and theirs status in a logical series.
I don't know if this diagram shows more info or not.

**Comment by nikku** [comment]
Both views show the same diagram, rendered by [bpmn-js](https://github.com/bpmn-io/bpmn-js). So I assume that it is more or less the same issue.

If we can make the diagram accessible to you, it will automaticallybe accessible in the Camunda Modeler, too (https://github.com/camunda/camunda-modeler/issues/2588).

**Comment by hamidrezaabroshan** [comment]
Hi again @nikku, I urgently need to work with the  Simple monitor and Camunda  modeler in my job. If possible, I want to have a session with tech people.
If possible, while the dev team is resolving these accessibility issues, I need some info to write a console client that gets me the info that I need at work.
Thanks.

**Comment by nikku** [comment]
@hamidrezaabroshan We'll not be able to follow up on this issue _immediately_.

For your specific needs, consider if the BPMN document or the [BPMN model](https://github.com/bpmn-io/bpmn-moddle), or the [diagram model](https://bpmn.io/toolkit/bpmn-js/walkthrough/#data-model) may provide you with the information you need.

Also consider to ping this back to simple-monitor. They may be able to work around this issue, i.e by providing accessibility information on top of or next to the diagram.

**Comment by hamidrezaabroshan** [comment]
@nikku , "Also consider to ping this back to simple-monitor. They may be able to work around this issue, i.e by providing accessibility information on top of or next to the diagram."
I created issue. What else can I do?

**Comment by hamidrezaabroshan** [comment]
Hi again @nikku  can you please give update if anything done regarding this issue?
